feat: Add Referral data type to JavaSDK

mridulgrc opened the pull request:
New data type Referral added to the JavaSDK and added a new thing type Task.

Relates to: CHBASE-7985

mridulgrc committed:
refact: Update java version

-Java version updated to 11 and added required dependencies.

(cherry picked from commit 42cb5df5c1318efb870eab5a8a46859159e6f8bd)

mridulgrc committed:
feat: Add Referral data type to JavaSDK

New data type Referral added to the JavaSDK and added a new thing type Task.

Relates to: CHBASE-7985

sachinskumar41 commented on chbase-jaxb/src/main/java/com/chbase/thing/oxm/jaxb/base/Task.java:
```diff
@@ -0,0 +1,258 @@
+	protected DateTime completedDate;
+	@XmlElement(name = "status-reason")
+	protected CodableValue statusReason;
+	@XmlElement(required = true)
```
It is optional. Remove required attribute

sachinskumar41 commented on chbase-jaxb/src/main/java/com/chbase/thing/oxm/jaxb/base/Task.java:
```diff
@@ -0,0 +1,258 @@
+// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.2.4-2 
+// See <a href="http://java.sun.com/xml/jaxb">http://java.sun.com/xml/jaxb</a> 
+// Any modifications to this file will be lost upon recompilation of the source schema. 
+// Generated on: 2013.05.07 at 02:23:59 PM PDT 
```
Correct the date

mridulgrc committed:
Refact: Remove auto generated lines

- Updated required fields.
- Removed unwanted namespace.

Relates to: CHBASE-7985

mridulgrc committed:
fix: Update set when date function

mridulgrc committed:
Revert "refact: Update java version"

This reverts commit fe94b1300a31d6bcdc855309ec550c63be2d5230.
